btk/survey.py

```diff
@@ -43,6 +47,84 @@
 }
 
 
+def define_synthetic_psf(mirror_diameter, effective_area, filt_wavelength, fwhm, atmospheric_model="Kolmogorov"):
+    """Credit: WeakLensingDeblending (https://github.com/LSSTDESC/WeakLensingDeblending)
+    Defines a synthetic galsim PSF model
+    Args:
+        mirror_diameter (float): in meters [m]
+        effective_area (float): effective total light collecting area in square meters [m2]
+        filt_wavelength (string): filter wavelength
+        fwhm (float): fwhm of the atmospheric component
+        atmospheric_model (string): type of atmospheric model
+    Returns:
+        psf_model: galsim psf model
+    """
+
+    # define atmospheric psf
+    if atmospheric_model == "Kolmogorov":
+        atmospheric_psf_model = galsim.Kolmogorov(fwhm=fwhm)
+    elif atmospheric_model == "Moffat":
+        atmospheric_psf_model = galsim.Moffat(2, fwhm=fwhm)
+    else:
+        raise NotImplementedError(
+            f"The atmospheric model request '{atmospheric_model}' is incorrect or not implemented."
+        )
+
+    # define optical psf if available
+    if mirror_diameter > 0:
+        mirror_area = np.pi * (0.5 * mirror_diameter) ** 2
+        area_ratio = effective_area / mirror_area
+        if area_ratio <= 0 or area_ratio > 1:
+            raise RuntimeError(
+                "Incompatible effective-area and mirror-diameter values."
+            )
+        obscuration_fraction = np.sqrt(1 - area_ratio)
+        lambda_over_diameter = 3600 * np.degrees(
+            1e-10 * filt_wavelength / mirror_diameter
+        )
+        optical_psf_model = galsim.Airy(
+            lam_over_diam=lambda_over_diameter, obscuration=obscuration_fraction
+        )
+        psf_model = galsim.Convolve(atmospheric_psf_model, optical_psf_model).withFlux(
+            1.0
+        )
+
+    else:
+        psf_model = atmospheric_model.withFlux(1.0)
+        
+    return psf_model
+
+
+def make_psf(survey, filt):
+    """Generate a PSF galsim model
+    Args:
+        survey (Survey): survey to use
+        filt (Filter): filter to use
+    Returns:
+        psf_model: galsim PSF model
+    """
+
+    assert((filt.fwhm==None) ^ (filt.psf_path==None))
```

Comment:
Does the '^' stand for xor? in that case I would just make a small suggestion:
```suggestion
    assert (filt.fwhm is None) ^ (filt.psf_path is None)
```

Reply:
also the XOR logic here behing the assert may need to be explicited in a comment (since it's not the most straightforward thing)

Reply:
Yes it is a XOR logic but indeed I will add a comment